include/rnp/rnp2.h

```diff
@@ -260,6 +264,54 @@ rnp_result_t rnp_key_have_public(rnp_key_handle_t key, bool *result);
 
 /* TODO define functions for password-based encryption */
 
+rnp_result_t rnp_op_sign_create(rnp_op_sign_t *op,
+                                rnp_ffi_t      ffi,
+                                rnp_input_t    input,
+                                rnp_output_t   output);
+
+rnp_result_t rnp_op_sign_set_signing_key(rnp_op_sign_t op, rnp_key_handle_t key);
+
+rnp_result_t rnp_op_sign_set_compression(rnp_op_sign_t op, const char *compression, int level);
+rnp_result_t rnp_op_sign_set_armor(rnp_op_sign_t op, bool armored, bool cleansign);
+rnp_result_t rnp_op_sign_set_detached(rnp_op_sign_t op, bool detached);
+
+rnp_result_t rnp_op_sign_set_timestamps(rnp_op_sign_t op,
+                                        uint32_t      creation_time,
+                                        uint32_t      expiration_time);
+rnp_result_t rnp_op_sign_set_hash_fn(rnp_op_sign_t op, const char *hash);
+rnp_result_t rnp_op_sign_set_file_name(rnp_op_sign_t op, const char *filename);
+rnp_result_t rnp_op_sign_set_file_mtime(rnp_op_sign_t op, uint32_t mtime);
+
+rnp_result_t rnp_op_sign_execute(rnp_op_sign_t op);
+rnp_result_t rnp_op_sign_destroy(rnp_op_sign_t op);
+
+rnp_result_t rnp_op_verify(rnp_op_verify_result_t *result,
```

Comment:
The verify API isn't really consistent with the create/execute/destroy convention that most of the other ops in here follow.

Reply:
True, but verify needs to return potentially a lot of different status information (verification info, hash fn used, etc - for each signature) so returning an array of verify struct seemed the best way to handle it. I am open to doing something else but don't see a clear approach.

Reply:
@dewyatt do you have in mind an alternative what verify should return -- or do we want to break it down further?

Reply:
Mostly I was just thinking that if it has `rnp_op` in the name, it should follow the same create/execute/destroy pattern that all the other `rnp_op` groups do.

Reply:
I would also stick to: 
- `rnp_op_verify_create(input, output)`
- `rnp_op_verify_detached_create(input, signature)`
- `rnp_op_verify_execute()`
The reason for this is difference of purpose and main parameters of detached and normal verification.
@dewyatt @ronaldtse What do you think?

Reply:
@ni4 Yep I think that makes sense. I guess similar changes would make sense on the sign op but that's up to you.

Reply:
@randombit Here is the main issue in this PR. What do you think about splitting methods in this way?